cf-upgrade/Makefile.am

```diff
@@ -21,23 +21,44 @@
 # (COSL) may apply to this file if you as a licensee so wish it. See
 # included file COSL.txt.
 #
-bin_PROGRAMS = cf-upgrade
+noinst_LTLIBRARIES = libcf-upgrade.la
```

Comment:
@karlhto This will require you to make a change in enterprise repo, and maybe buildscripts, see:

https://github.com/cfengine/core/pull/3367

And specifically:

https://github.com/cfengine/enterprise/commit/039511edc99267dd776ed40653375ef6db42a582

Reply:
Seems like `cf-upgrade` isn't present everywhere in buildscripts.

Reply:
@olehermanse Seems like `cf-upgrade` is not used at all in enterprise currently. It shouldn't be necessary then, right?

Reply:
> Seems like `cf-upgrade` isn't present everywhere in buildscripts.

If you search for cf-upgrade in buildscripts repo you can find many references to it. It's in all host/client packages, community and enterprise, but it's `rm`ed from hub packages.

> @olehermanse Seems like cf-upgrade is not used at all in enterprise currently. It shouldn't be necessary then, right?

See the `core-tools/Makefile.am` in the commit that I linked. When you are using this pattern, like the other binaries, you are compiling a library, and then only linking it to a binary if we are not building enterprise. If we are building enterprise the binaries are linked later, according to `core-tools/Makefile.am` in enterprise repo.

So, the result of this change is that cf-upgrade binary is only compiled and linked when we are building community, not when we are building enterprise. You need to add it to the Makefile.am which I pointed to in the comment above.